WalletWasabi/Crypto/ZeroKnowledge/Transcripting/Transcript.cs

```diff
@@ -44,7 +44,7 @@ private Transcript(byte[] hash)
 		/// <summary>
 		/// Absorb arbitrary data into the state.
 		/// </summary>
-		private Transcript AssociatedData(byte[] data) =>
+		private Transcript AssociateData(byte[] data) =>
```

Comment:
i'm familiar with this argument, and i agree in general, but still disagree here:

1. this API is described in the STROBE paper, this is bikeshedding an already established interface. for the bits I added i have no objections even though I didn't follow this style guide but borrowed STROBE's more declarative approach, but for method named after its API this is just needlessly distracting, consider that `System.Security.Cryptography` calls various functions we're using `SHA256` or `HMAC`, not `ComputeSha256` or `ComputeHmac`
2. "associated data" should be treated as  noun with no substructure, it's a term not a combination of an adjective and a noun. If you really insist on making it a verb the new name should be "AddAssociatedData" or something, because this is not about associating data with anything, rather it's adding associated data to a transcript

Reply:
ok my bad, that was strictly correct because it has classes, not functions, in the stdlib crypto and the method is called `ComputeHash`, i misremembered... but still, my point is that style conventions here are interfering with semantics here, they should not override.

Reply:
I think we don't need to stick letter by letter to the specification. I personally like the interface of the Rust's Transcript class:

| Rust | C# |
|----|-----|
| `append_message` | `AppendMessage` or `Add` | 
| `append_u64` |  `AppendUlong` or `Add` |
| `build_rng` | `Fork`? | 
| `challenge_bytes` | `GetChallenge` or `GenerateChallenge` |
| `new` | `ctor` |

I am NOT suggesting changing the names, just that the Merlin implementation uses names that do not match _exactly _ with the specs and how those same names could be translated to c# if one would have ported it.

Reply:
those are higher level operations, e.g. comitting a nonce etc.

the lower level operations should match STROBE's operations, I did change `AD` to `AssociatedData` because that means the same thing, but `AssociateData` really does not... https://github.com/dalek-cryptography/merlin/blob/master/src/strobe.rs#L76

Reply:
to be clear i don't really care about the method name per se, like i said, if it really needs a verb we can throw "add" in there... what i care bout is the term "associated data" being changed